tools/extract-debug-symbols.sh

```diff
@@ -13,26 +13,36 @@ STRIP="${STRIP:-strip}"
 
 function extract_linux_or_android {
 
-for input in $@ ; do
-	output="${input}${suffix}"	
+for input in "$@" ; do
+	output="${input}${suffix}"
+
+	# The --preserve-dates flag for strip and objcopy only has
+	# whole second resolution, so move to a new file and copy to
+	# preserve the timestamps
+	mv "$input" "$input.timestamps"
+	cp "$input.timestamps" "$input"
```

Comment:
I'd suggest using "touch -m -r "$input" "$input.timestamps" here. That should create the timestamp file with the modification time of the original without needing to copy the file.